### DISCUSSION TITLE
Reject unauthenticated mail a la Google

### DISCUSSION THREAD

**Discussion by koehn** [rspamd/rspamd#4641] [Ideas]
Google has started [rejecting “unauthenticated” email](https://www.spamresource.com/2023/09/gmail-now-rejecting-unauthenticated-mail.html), meaning email that doesn’t have valid SPF and doesn’t have valid DKIM. 

Is there a way to do this with Rspamd? I guess I’m trying to devise a rule that runs after the SPF and DKIM rules, and puts them together.

**Comment by fatalbanana** [comment]
It's a lousy idea by my estimation (in that I expect you would find yourself rejecting perfectly good mail).

The `AUTH_NA_OR_FAIL` composite in the stock configuration is relevant, you might up its scoring or act on it using [force_actions](https://rspamd.com/doc/modules/force_actions.html).

Alternative approach could be punishing auth failures for particular domains using [whitelist module](https://rspamd.com/doc/modules/whitelist.html).

**Comment by koehn** [comment]
I think Google will quickly force senders into compliance, leaving only spammers in non-compliance. 

Thanks for the `force_actions` tip; that looks like just the thing.

**Comment by PHPGangsta** [comment]
@koehn Since years I also think about "no auth, no entry". Google+Yahoo push it, which is good! But they only require this for "large senders" (`bulk senders — those who send more than 5,000 messages to Gmail addresses in one day —`). Small senders (private servers, small mailing lists) are not forced to do auth (not yet at least).

https://blog.google/products/gmail/gmail-security-authentication-spam-protection/

If you reject all unauthenticated mails in February (when Googles+Yahoos restriction kick in), you might loose important mails, like @fatalbanana said. You should definitely search your logs for `AUTH_NA`/`AUTH_NA_OR_FAIL` before you activate the `reject` action. You can do this in the `History` tab of the rspamd Web-GUI, or in your logs:

`grep "no action" rspamd.log | grep AUTH_NA`

If you don't find any false positives, you are lucky. I still get some important mails which do not authenticate, especially forwarded mails or mailinglist mails. I contacted the senders, and either they did not answer, or they said "auth is not needed, it works fine without it".

I currently give 7 score points for `AUTH_NA` and `AUTH_NA_OR_FAIL`, which is a good start. And if a sender does not authenticate, he still has the chance to deliver if everything else is flawless. Maybe in the future I'll reject and have a whitelist (with all false positives of the last 6 months or so), let's see...
